src/mlpack/tests/main_tests/local_coordinate_coding_test.cpp

```diff
@@ -0,0 +1,417 @@
+  arma::mat initDict = {{1, 1}, {2, 2}, {3, 3}};
+
+  SetInputParam("training", std::move(x));
+  SetInputParam("initial_dictionary", std::move(initDict));
+  SetInputParam("atoms", (int) 2);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Ensure that the number of columns in initial dictionary is same as 
+ * the number of atoms.
+ */
+BOOST_AUTO_TEST_CASE(LCCInitDictAtomTest)
+{
+  arma::mat x = {{1, 1, 1, 1}, {2, 2, 2, 2}, {3, 3, 3, 3}, {4, 4, 4, 4}};
+  arma::mat initDict = {{1, 1}, {2, 2}, {3, 3}, {4, 4}};
+
+  SetInputParam("training", std::move(x));
+  SetInputParam("initial_dictionary", std::move(initDict));
+  SetInputParam("atoms", (int) 3);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Ensure that training data and test data points
+ * have same dimensionality.
+ */
+BOOST_AUTO_TEST_CASE(LCCTrainAndTestDataDimTest)
+{
+  arma::mat x;
+  x.load("mnist_first250_training_4s_and_9s.arm");
+  arma::mat t = x;
+
+  t.shed_rows(1, 2);
+
+  // Input data.
+  SetInputParam("training", x);
+  SetInputParam("atoms", (int) 10);
+  SetInputParam("max_iterations", (int) 2);
+  SetInputParam("test", std::move(t));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Ensure that only one out of training and input_model are specified.
+ */
+BOOST_AUTO_TEST_CASE(LCCTrainAndInputModelTest)
+{
+  arma::mat x;
+  x.load("mnist_first250_training_4s_and_9s.arm");
+
+  SetInputParam("training", x);
+  SetInputParam("atoms", (int) 10);
+  SetInputParam("max_iterations", (int) 2);
+
+  mlpackMain();
+
+  LocalCoordinateCoding* outputModel =
+      std::move(CLI::GetParam<LocalCoordinateCoding*>("output_model"));
+
+  // No need to input training data again.
+  SetInputParam("input_model", std::move(outputModel));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Ensure that dimensionality of the trained model matches
+ * the dimensionality of the test points.
+ */
+BOOST_AUTO_TEST_CASE(LCCTrainedModelDimTest)
+{
+  arma::mat x;
+  x.load("mnist_first250_training_4s_and_9s.arm");
+  arma:: mat t = x;
+  t.shed_rows(1, 2);
+
+  SetInputParam("training", x);
+  SetInputParam("atoms", (int) 10);
+  SetInputParam("max_iterations", (int) 2);
+
+  mlpackMain();
+
+  LocalCoordinateCoding* outputModel =
+      std::move(CLI::GetParam<LocalCoordinateCoding*>("output_model"));
+
+  SetInputParam("input_model", std::move(outputModel));
+  SetInputParam("test", std::move(t));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/*
+* Ensure that the program throws error for negative number of atoms.
+*/
```

Comment:
Here's another one---what happens when the number of atoms is greater than the number of points in the dataset?

Reply:
I could not infer from the paper that the number of atoms has to be less than the number of data points but in general, we expect that it should be the case.